Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -97,6 +97,21 @@ internal final class DatadogCore {
         )
     }
 
+    /// Add or override the extra info of the current user
+    ///
+    ///  - Parameters:
+    ///    - extraInfo: The user's custom attibutes to add or override
+    func addUserExtraInfo(_ newExtraInfo: [AttributeKey: AttributeValue?]) {
+        var extraInfo = dependencies.userInfoProvider.value.extraInfo
+        for property in newExtraInfo {
+            if property.value == nil {
+                extraInfo.removeValue(forKey: property.key)
+            }
+        }
+        extraInfo.merge(newExtraInfo.compactMapValues { $0 }, uniquingKeysWith: { _, new in new })
```

Comment:
I think this can be done a in single line:
```suggestion
        newExtraInfo.forEach { extraInfo[$0.key] = $0.value }
```

Reply:
Nevermind... apparently you're right that adding nil to a dictionary that doesn't support it removes the key. TIL.

Reply:
if `$0.value` is nil it will remove it from `extraInfo`. I did test it on your branch, no compiler error and tests are passing :)